core/src/main/java/com/mapzen/android/graphics/ImportYamlGenerator.java

```diff
@@ -17,26 +21,42 @@
    */
   String getImportYaml(ThemedMapStyle themedMapStyle, int labelLevel, int detailLevel,
       ThemeColor color) {
-    String labelFileName = new StringBuilder()
-        .append("label-")
-        .append(labelLevel)
-        .append(".yaml")
-        .toString();
-    String detailFileName = new StringBuilder()
-        .append("detail-")
-        .append(detailLevel)
-        .append(".yaml")
-        .toString();
-    String colorFileName = new StringBuilder()
-        .append("color-")
-        .append(color.toString())
-        .append(".yaml")
-        .toString();
-    return "{ import: [ "
-        + themedMapStyle.getBaseStyleFilename() + ", "
-        + themedMapStyle.getThemesPath() + labelFileName + ", "
-        + themedMapStyle.getThemesPath() + detailFileName + ", "
-        + themedMapStyle.getThemesPath() + colorFileName
-        + " ] }";
+    StringBuilder importBuilder = new StringBuilder()
+        .append("{ import: [ ")
+        .append(themedMapStyle.getBaseStyleFilename())
+        .append(", ");
+
+    if (labelLevel != NONE) {
+      importBuilder.append(themedMapStyle.getThemesPath());
+      importBuilder.append("label-");
+      importBuilder.append(labelLevel);
+      importBuilder.append(".yaml");
+    }
+
+    if (labelLevel != NONE && (detailLevel != NONE || color != null)) {
+      importBuilder.append(", ");
+    }
+
+    if (detailLevel != NONE) {
+      importBuilder.append(themedMapStyle.getThemesPath());
+      importBuilder.append("detail-");
+      importBuilder.append(detailLevel);
+      importBuilder.append(".yaml");
+    }
+
+    if (detailLevel != NONE && color != null) {
+      importBuilder.append(", ");
+    }
+
+    if (color != null) {
+      importBuilder.append(themedMapStyle.getThemesPath());
+      importBuilder.append("color-");
+      importBuilder.append(color.toString());
+      importBuilder.append(".yaml");
+    }
+
+    importBuilder.append(" ] }");
+    Log.d("sarah", "IMPORT STR: " + importBuilder.toString());
```

Comment:
Hello sarah :D

I guess put for debugging purposes.

Reply:
:) Removed